astroid/interpreter/_import/spec.py

```diff
@@ -440,10 +441,15 @@ def find_spec(modpath: list[str], path: Sequence[str] | None = None) -> ModuleSp
     :return: A module spec, which describes how the module was
              found and where.
     """
+    return _find_spec(tuple(modpath), tuple(path) if path else None)
```

Comment:
nit: we can loosen the type annotation for modpath in this signature to Sequence

Reply:
Er, I guess both args could become Iterable

Reply:
Actually, there are a few mypy errors introduced in this work, would be nice to address them now:

```
astroid/interpreter/_import/spec.py:452: error: Incompatible types in assignment (expression has type "list[Any]", variable has type "tuple[Any, ...]")  [assignment]
astroid/interpreter/_import/spec.py:459: error: "tuple[Any, ...]" has no attribute "pop"  [attr-defined]
astroid/interpreter/_import/spec.py:461: error: Argument 3 to "_find_spec_with_path" has incompatible type "tuple[Any, ...]"; expected "list[str]"  [arg-type]
```

Reply:
Applied both suggestions, I had to change the argument name for mypy to stop complaining.